### PR TITLE
Fix flaky tests

### PR DESCRIPTION
**Pull request by jyu00**
### Summary
Addresses #580 . This probably doesn't fix all tests, but it's a start.

### CODE REVIEW

File: test/ibmq/websocket/test_websocket_integration.py [hunk 1]
```diff
@@ -142,16 +147,21 @@ def _job_status_side_effect(*args, **kwargs):
...
+        qobj = bell_in_qobj(backend, shots=backend.configuration().max_shots)
+        job = backend.run(qobj, validate_qobj=True)
 
-        with self.assertRaises(JobTimeoutError):
-            job.result(timeout=0.1)
+        try:
+            with self.assertRaises(JobTimeoutError):
+                job.result(timeout=0.1)
+        finally:
+            job.cancel()
```
Comment (luisg5) [inline test/ibmq/websocket/test_websocket_integration.py]: Might also be worth wrapping this in a try except, as mentioned above, so the test would not fail if there is a problem cancelling the job

File: test/utils.py [hunk 2]
```diff
@@ -53,3 +57,17 @@ def get_large_circuit(backend: IBMQBackend) -> QuantumCircuit:
     circuit.measure(list(range(n_qubits)), list(range(n_qubits)))
 
     return circuit
+
+
+def bell_in_qobj(backend: IBMQBackend, shots: int = 1024) -> Qobj:
```
Comment (luisg5) [inline test/utils.py]: Nice addition!

Comment (luisg5) [inline test/utils.py]: Nice addition!

File: test/utils.py [hunk 3]
```diff
@@ -71,3 +74,30 @@ def bell_in_qobj(backend: IBMQBackend, shots: int = 1024) -> Qobj:
...
+        verify: Verify job status.
+
+    Returns:
+        Whether job has been cancelled.
+    """
+    cancelled = False
+    for _ in range(2):
+        # Try twice in case job is not in a cancellable state
+        try:
+            if job.cancel():
```
Comment (luisg5) [inline test/utils.py]: You forgot to set `cancelled = True` within `if job.cancel()` 😅

Comment (jyu00) [reply test/utils.py]: Oops

**Pull request merged**